parachain/primitives/core/src/operating_mode.rs

```diff
@@ -0,0 +1,82 @@
+use codec::{Decode, Encode, FullCodec, MaxEncodedLen};
+use frame_support::{
+	dispatch::DispatchResult, traits::EnsureOrigin, PalletError, RuntimeDebug, StorageValue,
+};
+use scale_info::TypeInfo;
+use sp_std::fmt::Debug;
+
+/// Error generated by the `OwnedBridgeModule` trait.
+#[derive(Encode, Decode, TypeInfo, PalletError)]
+pub enum BridgeModuleError {
+	/// All pallet operations are halted.
+	Halted,
+}
+
+/// Operating mode for a bridge module.
+pub trait OperatingMode: Send + Copy + Debug + FullCodec {
+	// Returns true if the bridge module is halted.
+	fn is_halted(&self) -> bool;
+}
+
+/// Basic operating modes for a bridges module (Normal/Halted).
+#[derive(
+	Encode,
+	Decode,
+	Clone,
+	Copy,
+	PartialEq,
+	Eq,
+	RuntimeDebug,
+	TypeInfo,
+	MaxEncodedLen,
+	serde::Serialize,
+	serde::Deserialize,
+)]
+pub enum BasicOperatingMode {
```

Comment:
Lets rename this to `OperatingMode`, and then rename the trait `OperatingMode` to `OperatingModeTrait`.

Reply:
https://github.com/Snowfork/snowbridge/pull/969/commits/09f213337268a6f924eff1c4e556d951f210f729